merlin/core/protocols.py

```diff
@@ -0,0 +1,116 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class DictLike(Protocol):
+    def __iter__(self):
+        return iter([])
```

Comment:
The actual implementation of these methods don't matter, since they'll be overridden by anything that explicitly implements the protocol and otherwise are only used to check that matching method signatures are available on anything compared with `isinstance()`. These fake implementations make the linters happy though, since they insist that `__iter__()` must return an iterator and `__len__` must return a non-negative integer.